### PR TITLE
Allow full ResourcePool path

### LINKED ISSUE #118: Configured resource pool not found when using child resource pool
When cloning from template, **vsphere.resource_pool_name** has to be configured properly. When I use child resource pool such as:

    vsphere.resource_pool_name = 'Technology/Applied Research and Development/Test Automation'

I get error **Configured resource pool not found**
But when using root resource pool, it's ok.

    vsphere.resource_pool_name = 'Technology'

Is there any other way how to configure child resource pool?

### PR DESCRIPTION
**Pull request by davidhrbac**
Fixes #118 

You can specify resource pool like this:

```
vsphere.resource_pool_name = 'RPs/RP1/ChildRP'
```

### CODE REVIEW

File: lib/vSphere/util/vim_helpers.rb [hunk 1]
```diff
@@ -13,12 +13,22 @@ def get_vm_by_uuid(connection, machine)
...
         def get_resource_pool(datacenter, machine)
-          compute_resource = get_compute_resource(datacenter, machine)
-          rp = compute_resource.resourcePool
-          unless machine.provider_config.resource_pool_name.nil?
-            rp = compute_resource.resourcePool.find(machine.provider_config.resource_pool_name)
-            fail Errors::VSphereError, :missing_resource_pool if rp.nil?
+          rp = get_compute_resource(datacenter, machine)
+          entity_array = machine.provider_config.resource_pool_name.split('/')
+          entity_array.each do |entity_array_item|
+            puts entity_array_item
```
Comment (rylarson) [inline lib/vSphere/util/vim_helpers.rb]: This must look odd in the output. No news is good news. We shouldn't be outputting each item in the array

File: lib/vSphere/util/vim_helpers.rb [hunk 2]
```diff
@@ -13,12 +13,22 @@ def get_vm_by_uuid(connection, machine)
...
+            next if entity_array_item == ''
```
Comment (rylarson) [inline lib/vSphere/util/vim_helpers.rb]: `entity_array_item.empy?`

**Comment by rylarson** [comment]
I see that this change is almost identical to the way [knife-vsphere](https://github.com/chef-partners/knife-vsphere/blob/e1a81732b276071412408eca35cb4979555c609d/lib/chef/knife/base_vsphere_command.rb#L205) solves this problem, so it looks pretty good.

Probably need some unit tests here and to find out why this change broke the build.

**Comment by davidhrbac** [comment]
Made corrections based on your comments. As to knife-vshepre I was not aware of that project. Well, it seems we both came from the very same source.

**Comment by davidhrbac** [comment]
@rylarson Have you got the time to check the issue with the tests?

**Comment by lukastopiarz** [comment]
Works OK.

**Comment by davidhrbac** [comment]
@rylarson any progress on accepting? We have just helped another colleague of mine with this patch to have vagrant VM in a specific pool.

**Comment by rylarson** [comment]
@davidhrbac 

I don't have write access so I can't merge it, I just happen to do a lot of work on the plugin. Have you run the tests locally? Most of the tests set up rather complicated mocks (because they have to), and changing things about the types that get returned and what not often breaks them.

That is probably what is going on here.

**Comment by michael-brandt-cu** [comment]
@davidhrbac I've been looking a bit at the tests that are failing, and it looks like it's more than a mocking problem; since you've actually changed the function, the tests that depended on the old functionality are failing.

Unfortunately I wasn't working on this project at the beginning, and I'm having trouble understanding each of the tests that are failing, and how exactly it relates to your changes. I'll keep looking at it when I have the time to.

And the "translation missing" thing you reference in be08244, 00f4244, and 6a994d3 refers to `locale/en.yml`. I think just need to add the right key under `en.errors` with your desired error text.

**Pull request closed without merging**